avogadro/io/lammpsformat.cpp

```diff
@@ -0,0 +1,413 @@
+      x_min = lexicalCast<double>(box_bounds_x.at(0));
+      x_max = lexicalCast<double>(box_bounds_x.at(1));
+      // Read y_min, y_max
+      getline(inStream, buffer);
+      vector<string> box_bounds_y(split(buffer, ' '));
+      y_min = lexicalCast<double>(box_bounds_y.at(0));
+      y_max = lexicalCast<double>(box_bounds_y.at(1));
+      // Read z_min, z_max
+      getline(inStream, buffer);
+      vector<string> box_bounds_z(split(buffer, ' '));
+      z_min = lexicalCast<double>(box_bounds_z.at(0));
+      z_max = lexicalCast<double>(box_bounds_z.at(1));
+    }
+
+    // x,y,z stand for the coordinate axes
+    // s stands for scaled coordinates
+    // u stands for unwrapped coordinates
+    // scale_x = 0. if coordinates are cartesian and 1 if fractional (scaled)
+    getline(inStream, buffer);
+    labels = vector<string>(split(buffer, ' '));
+    for (size_t i = 0; i < labels.size(); i++) {
+      if (labels[i] == "x" || labels[i] == "xu") {
+        x_idx = i;
+        scale_x = 0.;
+      } else if (labels[i] == "xs" || labels[i] == "xsu") {
+        x_idx = i;
+        scale_x = 1.;
+      } else if (labels[i] == "y" || labels[i] == "yu") {
+        y_idx = i;
+        scale_y = 0.;
+      } else if (labels[i] == "ys" || labels[i] == "ysu") {
+        y_idx = i;
+        scale_y = 1.;
+      } else if (labels[i] == "z" || labels[i] == "zu") {
+        z_idx = i;
+        scale_z = 0.;
+      } else if (labels[i] == "zs" || labels[i] == "zsu") {
+        z_idx = i;
+        scale_z = 1.;
+      } else if (labels[i] == "type")
+        type_idx = i;
+      else if (labels[i] == "id")
+        id_idx = i;
+    }
+
+    Array<Vector3> positions;
+    positions.reserve(numAtoms);
+
+    for (size_t i = 0; i < numAtoms; ++i) {
+      getline(inStream, buffer);
+      vector<string> tokens(split(buffer, ' '));
+      if (tokens.size() < 5) {
+        appendError("Not enough tokens in this line: " + buffer);
+        return false;
+      }
+      // If parsed coordinates are fractional, the corresponding unscaling is
+      // done. Else the positions are assigned as parsed.
+      Vector3 pos(
+        (1 - scale_x) * lexicalCast<double>(tokens[x_idx - 2]) +
+          scale_x *
+            (x_min + (x_max - x_min) * lexicalCast<double>(tokens[x_idx - 2])),
+        (1 - scale_y) * lexicalCast<double>(tokens[y_idx - 2]) +
+          scale_y *
+            (y_min + (y_max - y_min) * lexicalCast<double>(tokens[y_idx - 2])),
+        (1 - scale_z) * lexicalCast<double>(tokens[z_idx - 2]) +
+          scale_z *
+            (z_min + (z_max - z_min) * lexicalCast<double>(tokens[z_idx - 2])));
+      positions.push_back(pos);
+    }
+
+    mol.setCoordinate3d(positions, coordSet++);
+    mol.setUnitCell(new UnitCell(Vector3(x_max - x_min, 0, 0),
+                                 Vector3(tilt_xy, y_max - y_min, 0),
+                                 Vector3(tilt_xz, tilt_yz, z_max - z_min)));
+
+    positions.clear();
```

Comment:
Is this needed? It looks to me like it is about to go out of scope, and so would be cleared, and then a new one created on the next loop.